### PR TITLE
Create reconcile test for azuremanagedcluster_controller

### LINKED ISSUE #4270: Add unit test to `AzureManagedClusterReconciler`
/kind cleanup

Add unit test coverage for azuremanagedcluster_reconciler.go. This is a part of a larger effort to improve overall test coverage in the controllers package: https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/3649

### PR DESCRIPTION
**Pull request by troy0820**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Helps to resolve issue with not having enough test coverage.  

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4270

**Special notes for your reviewer**:
None

- [ ] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [x] adds unit tests

**Release note**:
```release-note
None
```

### CODE REVIEW

File: controllers/azuremanagedcluster_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2020 The Kubernetes Authors.
```
Comment (mboersma) [inline controllers/azuremanagedcluster_controller_test.go:2]:
```suggestion
Copyright 2024 The Kubernetes Authors.
```

New files always get the current year for copyright, is my understanding.

File: controllers/azuremanagedcluster_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func TestAzureManagedClusterController(t *testing.T) {
+	g := NewWithT(t)
+	ctx := context.Background()
+	scheme := runtime.NewScheme()
+	err := infrav1.AddToScheme(scheme)
```
Comment (mboersma) [inline controllers/azuremanagedcluster_controller_test.go:41]:
```suggestion
	g.Expect(infrav1.AddToScheme(scheme)).To(Succeed())
```
This is equivalent to these two lines, but is a bit more idiomatic for Ginkgo IMHO.

File: controllers/azuremanagedcluster_controller_test.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	g.Expect(err).ToNot(HaveOccurred())
+	err = clusterv1.AddToScheme(scheme)
```
Comment (mboersma) [inline controllers/azuremanagedcluster_controller_test.go:43]:
```suggestion
	g.Expect(clusterv1.AddToScheme(scheme)).To(Succeed())
```

File: controllers/azuremanagedcluster_controller_test.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+			},
+		},
+	}
+	aksCluster := &infrav1.AzureManagedCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fake-cluster",
+			Namespace: "fake-namespace",
+		},
+	}
+	err = controllerutil.SetOwnerReference(cluster, aksCluster, scheme)
```
Comment (mboersma) [inline controllers/azuremanagedcluster_controller_test.go:74]:
```suggestion
	g.Expect(controllerutil.SetOwnerReference(cluster, aksCluster, scheme)).To(Succeed())
```

File: controllers/azuremanagedcluster_controller_test.go [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	}
+	to := reconciler.Timeouts{}
+	rec := &AzureManagedClusterReconciler{
+		Timeouts: to,
+		Client:   fakeclient,
+	}
+
+	_, err = rec.Reconcile(ctx, req)
+	g.Expect(err).ToNot(HaveOccurred())
+	err = rec.Get(ctx, client.ObjectKeyFromObject(aksCluster), aksCluster)
```
Comment (mboersma) [inline controllers/azuremanagedcluster_controller_test.go:90]:
```suggestion
	g.Expect(rec.Get(ctx, client.ObjectKeyFromObject(aksCluster), aksCluster)).To(Succeed())
```

**Review by mboersma — COMMENTED** [review]
This is great @troy0820, thanks! I just had some trivial style comments.

**Review by mboersma — APPROVED** [review]
lgtm

Could you squash the 2 commits please?

**Comment by troy0820** [comment]
@mboersma squashed and force pushed.

**Review by mboersma — APPROVED** [review]
/lgtm

**Review by willie-yao — APPROVED** [review]
/lgtm
/approve

Thanks for working on this!! @troy0820

**Pull request merged**